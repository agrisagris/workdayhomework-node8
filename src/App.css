.container button {
  padding: 10px 20px;
  border-radius: 10px;
  margin-top: 10px;
  border-style: none;
  background-color: #f0f4f8;
}
.container button:hover {
  background-color: #d9e2ec;
}

.modal-content, .container button {
  box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06);
}

.container {
  padding: 20px;
}
.container div {
  font-weight: bold;
}
.modal-overlay {
  background-color: rgba(0, 0, 0, 0.15);
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.modal-content {
  margin: auto;
  margin-top: 25vh;
  width: 200px;
  padding: 20px;
  background-color: white;
  border-radius: 6px;
}

table {
  margin-top: 20px;
  width: 300px;
}
table th {
  background: #d9e2ec;
  padding: 2px 10px;
}
table td {
  font-family: monospace;
  padding: 2px 10px;
}
table td:nth-child(even) {
  text-align: right;
}
table td:nth-child(odd) {
  text-align: center;
}
table tr:nth-child(even) {
  background: #f0f4f8;
}

/*# sourceMappingURL=App.css.map */
