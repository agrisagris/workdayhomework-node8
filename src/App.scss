%button {
  padding: 10px 20px;
  border-radius: 10px;
  margin-top: 10px;
  border-style: none;
  background-color: #f0f4f8;
  @extend %shadow;
  &:hover {
    background-color: #d9e2ec;
  }
}

%shadow {
  box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1),
    0 2px 4px -1px rgba(0, 0, 0, 0.06);
}

.container {
  padding: 20px;
  div {
    font-weight: bold;
  }
  button {
    @extend %button;
  }
}

.modal-overlay {
  background-color: rgba(0, 0, 0, 0.15);
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.modal-content {
  margin: auto;
  margin-top: 25vh;
  width: 200px;
  padding: 20px;
  background-color: white;
  border-radius: 6px;
  @extend %shadow;
}

table {
  margin-top: 20px;
  width: 300px;
  th {
    background: #d9e2ec;
    padding: 2px 10px;
  }
  td {
    font-family: monospace;
    padding: 2px 10px;
  }
  td:nth-child(even) {
    text-align: right;
  }
  td:nth-child(odd) {
    text-align: center;
  }
  tr:nth-child(even) {
    background: #f0f4f8;
  }
}
