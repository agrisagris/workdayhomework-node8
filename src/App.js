import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";

import * as ratesActions from "./redux/actions/ratesActions";
import ratesSelector from "./redux/selectors/ratesSelector";
import RatesTable from "./components/RatesTable";
import "./App.css";

export default function App() {
  const dispatch = useDispatch();
  const rates = useSelector(ratesSelector);
  const [showDialog, setShowDialog] = useState(false);

  const toogleShowDialog = () => {
    setShowDialog(!showDialog);
  };
  const onLoadRates = () => {
    dispatch(ratesActions.loadRates());

    toogleShowDialog();
  };
  return (
    <div className="container">
      <div>Here be dragons</div>
      <button type="button" onClick={onLoadRates}>
        Load Rates
      </button>
      <RatesTable
        rates={rates}
        showDialog={showDialog}
        toogleShowDialog={toogleShowDialog}
      />
    </div>
  );
}
