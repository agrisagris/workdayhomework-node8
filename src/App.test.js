import React from "react";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { Provider as ReduxProvider } from "react-redux";
import thunk from "redux-thunk";

import App from "./App";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const component = () =>
  render(
    <ReduxProvider store={mockStore({})}>
      <App />
    </ReduxProvider>
  );

test("renders todo", () => {
  const { getByText } = component();
  const todoElement = getByText(/here be dragons/i);

  expect(todoElement).toBeInTheDocument();
});

test("Load rates button:renders", () => {
  const { getByRole } = component();

  const buttonElement = getByRole("button", { name: /load rates/i });

  expect(buttonElement).toBeInTheDocument();
});

test("Load rates button:snapshot before press", () => {
  const { container } = component();

  expect(container).toMatchSnapshot();
});
