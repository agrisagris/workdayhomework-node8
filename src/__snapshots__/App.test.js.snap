// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load rates button:snapshot before press 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      Here be dragons
    </div>
    <button
      type="button"
    >
      Load Rates
    </button>
    
  </div>
</div>
`;
